pkgs/servers/dns/knot-resolver/default.nix

```diff
@@ -17,36 +17,32 @@ inherit (lib) optional optionals optionalString;
 lua = luajitPackages;
 
 unwrapped = stdenv.mkDerivation rec {
-  pname = "knot-resolver";
-  version = "5.7.4";
-
-  src = fetchurl {
-    url = "https://secure.nic.cz/files/knot-resolver/${pname}-${version}.tar.xz";
-    hash = "sha256-a22m7PBoKAQa+tRN+iJ3gfCuNK0YOmZwCFCTVdGL2cg=";
+  pname = "knot-resolver-core";
+  version = "6.0.8-" + src.rev;
+
+  src = fetchFromGitLab {
+    domain = "gitlab.nic.cz";
+    owner = "knot";
+    repo = "knot-resolver";
+    fetchSubmodules = true; # yes, unfortunately hard to work around
+    #rev = "7ef1a7ce0d4"; hash = "sha256-uWoXBgZV42DoBLTZ27gAuIXw7c6iJlAYsOsGhu0Gz+Y=";
+    rev = "b8ec0127"; hash = "sha256-16xFNMsSg+6cZLhtk9BlW5skwP7IDKrFEYZNpoxbJMY="; # !1613 @ 2024-09-24
   };
 
-  outputs = [ "out" "dev" ];
+  #dontStrip = true; # FIXME: TMP
+
+  outputs = [ "out" "dev" "config_py" ];
 
   # Path fixups for the NixOS service.
   postPatch = ''
     patch meson.build <<EOF
-    @@ -50,2 +50,2 @@
+    @@ -50,2 +50,3 @@
```

Comment:
Lets move that into its own file

Reply:
It might be temporary.  We're now discussing upstream if we can have a more clever default, covering more systems on its own.